### ISSUE TITLE
[iOS] Vertical CollectionView does not show simple Items

### ISSUE THREAD

**Issue by Phenek** [dotnet/maui#8489]
### Description

Hello, I'm testing MAUI about Collection, ListView scroll etc...,
When I try to display a simple View as follow in CollectionView
````
<ContentView>
       <Frame BackgroundColor="{Binding .}" Margin="10,5,10,5" HeightRequest="300" CornerRadius="5"/>
</ContentView>
````
CollectionView does not show any items and I don't have any exception or warnings.
But the same views render well in ListView 

![Group 448](https://user-images.githubusercontent.com/4518686/177007228-443b4e76-da05-4c4c-a63b-488d08046f41.png)

### Steps to Reproduce

You can download my workplace project and run it.
[ScrollConflicts.zip](https://github.com/dotnet/maui/files/8869797/ScrollConflicts.zip)
It display one CollectionView and one ListView, rendering the same type of views

### Version with bug

6.0.312

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15

**Comment by Redth** [comment]
I made a few changes to the repro and it is working well on iOS:

1. Upgrade project to net7
2. Remove the extra `<ContentView>` in tab1's Item template - it's unnecessary and not set to fill the space which is probably the main issue here.
3. Switch Frame usage to Border
4. Removed an extra Stack Layout surrounding the root carouselview on the page - again, unnecessary nesting.

![simulator_screenshot_90A5CE5E-FDD6-4B94-9B69-D85BA3F84986](https://user-images.githubusercontent.com/271950/204358730-e2b78308-f857-40b6-90dc-7935ad330afe.png)
